assets/js/theme/common/product-details.js

```diff
@@ -35,8 +35,23 @@ export default class Product {
         this.imageGallery.init();
         this.listenQuantityChange();
         this.initRadioAttributes();
-        this.updateProductAttributes(productAttributesData);
-        this.updateView(productAttributesData);
+
+        const $form = $('form[data-cart-item-add]', $scope);
+        const hasOptions = $('[data-product-option-change]', $form).html().trim().length;
+
+        // Update product attributes. If we're in quick view and the product has options, then also update the initial view in case items are oos
+        if (_.isEmpty(productAttributesData) && hasOptions && window.FormData !== undefined) {
```

Comment:
seems like jquery.serialize() does not use FormData. So you can remove that check